### PR TITLE
feat: Migrating Offers related components to /snek

### LINKED ISSUE #3888: Migrating Offers related components to /snek
Since our /snek is up and running, we'd like to bring over the already implemented functionality from /bsx, I may have forgotten something + I'm not sure about the indexer here, whether new resolvers need to be written or not, regardless. this needs to be done. In case of required indexer work, this can be eventually split into multiple issues.

In the meantime, it would be nice to generate some offer activity on /snek so we have something to test this on ( ideally mix in some offers with lower expiration dates as well 👀 - @JustLuuuu @DeluneClaire @cryptodamsky )

The stuff we're missing: 

1. routes to /offers and /stats
2. make sure routes work from navbar:
image
3. Offers Made tab at user profile:
image
4. Incoming offers for user /incomingoffers
image

### PR DESCRIPTION
**Pull request by zhengow**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3888 
- [ ] Requires deployment <rubick/magick_issue>
- [x] <brief_description_of_what_I've_added>
  1. routes to /offers and /stats
  2. Offers Made tab at user profile:
  3. Incoming offers for user /incomingoffers

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image
image
image
image

### CODE REVIEW

File: components/rmrk/Profile/NavbarProfileDropdown.vue [hunk 1]
```diff
@@ -45,8 +45,8 @@
           <nuxt-link to="/transform">{{ $t('transform') }}</nuxt-link>
         </b-dropdown-item>
       </template>
-      <b-dropdown-item v-if="isBsx" has-link aria-role="menuitem">
-        <nuxt-link to="/bsx/incomingoffers">{{
+      <b-dropdown-item v-if="isBsx || isSnek" has-link aria-role="menuitem">
```
Comment (kkukelka) [inline components/rmrk/Profile/NavbarProfileDropdown.vue:48]: can we make this just 1 prop called "showIncommingOffers"
and in `Navbar.vue` just propagate it via `:show-incomming-offers="isBsx || isSnek"`

File: components/rmrk/Profile/NavbarProfileDropdown.vue [hunk 2]
```diff
@@ -99,6 +99,7 @@ export default class NavbarProfileDropdown extends mixins(PrefixMixin) {
   @Prop() public value!: any
   @Prop() public isRmrk!: boolean
   @Prop() public isBsx!: boolean
+  @Prop() public isSnek!: boolean
```
Comment (kkukelka) [inline components/rmrk/Profile/NavbarProfileDropdown.vue:102]: so let's combine them

**Review by kkukelka — CHANGES_REQUESTED** [review]

**Comment by petersopko** [comment]
@zhengow let's make sure the displayed currency matches everywhere (should be KSM, not BSX)
examples I quickly found:
image
and
image

**Comment by kkukelka** [comment]
+ 2 new conflicts

**Comment by petersopko** [comment]
@zhengow if #3884 gets merged before this one, please uncomment some of the navbar tests in `basicE2E.cy.ts` -> they were disabled until we get stats back in navbar

**Comment by zhengow** [comment]
> * 2 new conflicts

resolved:)

**Comment by zhengow** [comment]
> @zhengow let's make sure the displayed currency matches everywhere (should be KSM, not BSX) examples I quickly found: image and image

Fix it by looking at #3906

**Comment by zhengow** [comment]
> @zhengow if #3884 gets merged before this one, please uncomment some of the navbar tests in `basicE2E.cy.ts` -> they were disabled until we get stats back in navbar

There is no commented navbar tests now, is now ok?

**Comment by zhengow** [comment]
> @zhengow if #3884 gets merged before this one, please uncomment some of the navbar tests in `basicE2E.cy.ts` -> they were disabled until we get stats back in navbar

Done,
image

**Review by kkukelka — APPROVED** [review]
all good now

**Review by petersopko — APPROVED** [review]
wfm

**Comment by petersopko** [comment]
some leftovers with BSX->KSM as displayed currency, can be possibly solved in:
- https://github.com/kodadot/nft-gallery/pull/3906

**Comment by petersopko** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 49.8 USD/KSM ~ 1.004 $KSM
🧗 [EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws](https://kusama.subscan.io/account/EgCzZLpVe7zg7RHh79qPyC8UrykVkA22YY2oj7sWMZf15Ws)
🔗 [0x7dc5bf5ab3055ea883eb4e4d9dcf34420e47c1454206e57d31ca8cc6c677783c](https://kusama.subscan.io/extrinsic/0x7dc5bf5ab3055ea883eb4e4d9dcf34420e47c1454206e57d31ca8cc6c677783c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**